regression/test.pl

```diff
@@ -352,4 +355,38 @@ ($)
 
 close LOG;
 
+if($opt_p && $failures != 0) {
+  open LOG,"<tests.log" or die "Failed to open tests.log\n";
+
+  my $printed_this_test = 1;
+  my $current_test = "";
+  my $output_file = "";
+  my $descriptor_file = "";
+
+  while (<LOG>) {
+    chomp;
+    if (/^Test '(.+)'/) {
+      $current_test = $1;
+      $printed_this_test = 0;
+    } elsif (/Descriptor:\s+([^\s]+)/) {
+      $descriptor_file = $1;
+    } elsif (/Output:\s+([^\s]+)/) {
+      $output_file = $1;
+    } elsif (/\[FAILED\]\s*$/) {
+      if(0 == $printed_this_test) {
+        $printed_this_test = 1;
+        print "\n\n";
+        print "Failed test: $current_test\n";
+        open FH, "<$current_test/$output_file";
+        while (my $f = <FH>) {
+          print $f;
+        }
+        close FH;
+        print "\n\nFailed $descriptor_file lines:\n";
+      }
+      print "$_\n";
```

Comment:
My perl isn't ~very good~ existent, but it might be more readable for non-perl users to `chomp $line;` and `print $line`. Also how come it isn't inside the `if(0 == $printed_this_test)`. (I realise this is code moved from `failed-tests-printer.pl` so feel free to say this is out of scope).